### ISSUE TITLE
Package to help working with filepaths

### ISSUE THREAD

**Issue by 1danjordan** [ropensci/unconf17#99]
Working with filepaths in R is a total pain in the ass, mainly because we treat them exclusively as character vectors. This means we're stuck parsing filepaths with messy regexes at best, and still no way of reliably building filepaths apart from string concatenation. There's also no way of building relative and absolute paths reliably.

I propose a package with a `Path` object that provides us structured filepaths and methods in a similar way to [Ammonite's Path object](https://github.com/lihaoyi/Ammonite/blob/master/ops/src/main/scala/ammonite/ops/Path.scala) in Scala. For some concrete examples of this, you can see Li Haoyi demonstrating it at minute 18 of his [A Better Scala REPL talk](https://www.youtube.com/watch?v=dP5tkmWAhjg).

This package could have implementations of `list.files`, `list.dirs`, `getwd` and so on, or `Path` could provide a factory method for parsing their outputs so we could still use them like `list.files() %>% Path`. 

I believe the current state of filepaths is a real pain point, and think a solution like this would be a good direction, but am obviously open to discussing other solutions!

**Comment by gaborcsardi** [comment]
Good idea. Here is some (incomplete) preliminary work: https://github.com/richfitz/pathr

**Comment by 1danjordan** [comment]
Another advantage would be handling paths regardless of OS - escaping Windows paths or flipping them is driving me nuts on my work machine! 😒
